aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -691,7 +696,60 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
         onSuccess: Consumer<MutableList<AuthUserAttribute>>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        authStateMachine.getCurrentState { authState ->
+            when (authState.authNState) {
+                // Check if user signed in
+                is AuthenticationState.SignedIn -> {
+
+                    GlobalScope.launch {
+                        try {
+                            var accessToken = getAccessToken()
+                            val getUserRequest = GetUserRequest.invoke {
+                                this.accessToken = accessToken
+                            }
+                            val user = configureCognitoClients().cognitoIdentityProviderClient?.getUser(getUserRequest)
+                            val userAttributes = buildList {
+                                user?.userAttributes?.mapTo(this) {
+                                    AuthUserAttribute(
+                                        AuthUserAttributeKey.custom(it.name),
+                                        it.value
+                                    )
+                                }
+                            }
+                            onSuccess.accept(userAttributes.toMutableList())
+                        } catch (e: Exception) {
+                            onError.accept(AuthException.UnknownException(e))
```

Comment:
Please use `CognitoExceptionConvertor` to also capture service exceptions.

Reply:
shouldn't the default exception capture all other exceptions?